### PR TITLE
test(core): add parity testing module and fix CLI test isolation

### PR DESCRIPTION
**Pull request by galligan**
Extract GitHub parity comparison logic from verify script into
testable core module. Add comprehensive unit tests for parity
comparison including edge cases for missing/extra threads.

Also adds --offline flag to fb command for local-only operations,
fixing tests that previously required GitHub auth for read-only
queries.

Test fixes:
- Add XDG_* environment variable overrides in CLI tests for
  proper isolation from host environment
- Update deprecated flag names (--json→--jsonl, --prs→--pr)

Co-Authored-By: Claude Opus 4.5 <noreply@anthropic.com>

### CODE REVIEW

File: apps/cli/src/commands/fb.ts [hunk 1]
```diff
@@ -570,19 +572,25 @@ async function createContext(options: FbCommandOptions): Promise<FbContext> {
...
   const repo = await resolveRepoOrThrow(options.repo);
   const { owner, name } = parseRepoInput(repo);
+  const offline = options.offline ?? false;
 
-  const auth = await detectAuth(config.github_token);
-  if (!auth.token) {
-    throw new Error(auth.error);
+  let client: GitHubClient | null = null;
+  if (!offline) {
+    const auth = await detectAuth(config.github_token);
```
[the remark this answers is not in the document]
Comment (galligan) [reply apps/cli/src/commands/fb.ts]: Fixed - added requireOnlineClient validation before all write operations.
